src/test/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImplTest.java

```diff
@@ -228,5 +230,66 @@ public void testFromProjectWithDockerAlfrescoPlugin() {
         assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("SUB_PROJECT1_DOCKER_IMAGE"));
     }
 
+    @Test
+    public void testFromProjectWithEnvironmentVariable(){
+        composeConvention.fromProject("testname",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME_BUILD_DOCKER_IMAGE"));
+    }
+
+    @Test
+    public void testFromProjectWithEnvironmentVariable1(){
+        TaskProvider<DockerBuildImage> dockerBuildImageProvider = project.getTasks()
+                .register("buildDockerImageAlternative", DockerBuildImage.class);
+        dockerBuildImageProvider.configure(dockerBuildImage -> {
+            dockerBuildImage.getImageId().set(DOCKER_IMAGE_ID);
```

Comment:
Can you give this task another image ID? (Just make up a random string)

That way, we can make sure that no two tasks get mixed up somewhere.